pandas/tests/frame/test_arithmetic.py

```diff
@@ -2132,3 +2132,12 @@ def test_enum_column_equality():
     expected = Series([True, True, True], name=Cols.col1)
 
     tm.assert_series_equal(result, expected)
+
+
+def test_mixed_col_index_dtype():
+    # GH 47382
+    df1 = DataFrame(columns=list("abc"), data=0.0, index=[0])
+    df2 = DataFrame(columns=list("abc"), data=0.0, index=[0])
+    df1.columns = df2.columns.astype("string")
+    df3 = df1 + df2
+    assert df3.columns.dtype == "object"
```

Comment:
Could you construct an expected DataFrame result and use `tm.assert_frame_equal(result, expected)`?

Reply:
Of course! Just committed that modification.